src/components/Shell/RoomShareDialog.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  Alert,
+  AlertColor,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  FormControlLabel,
+  TextField,
+  Tooltip,
+} from '@mui/material'
+import { useEffect, useState } from 'react'
+import { encodePassword } from 'utils'
+
+export interface RoomShareDialogProps {
+  roomId: string
+  isOpen: boolean
+  handleClose: () => void
+  copyToClipboard: (
+    content: string,
+    alert: string,
+    severity: AlertColor
+  ) => Promise<void>
+}
+
+export function RoomShareDialog(props: RoomShareDialogProps) {
+  const [password, setPassword] = useState('')
+  const [isAdvanced, setIsAdvanced] = useState(false)
+  const [isUnderstood, setIsUnderstood] = useState(false)
+  const handleClose = () => {
+    props.handleClose()
+    setPassword('')
+  }
+
+  useEffect(() => {
+    if (!isUnderstood) setPassword('')
+  }, [isUnderstood])
+
+  useEffect(() => {
+    if (!isAdvanced) setIsUnderstood(false)
+  }, [isAdvanced])
+
+  const url = window.location.href.split('#')[0]
+
+  const copyWithPass = async () => {
+    const secret = await encodePassword(props.roomId, password)
+    const params = new URLSearchParams()
+    params.set('secret', secret)
+    await props.copyToClipboard(
+      url + '#' + params.toString(),
+      'Private room URL with password copied to clipboard',
+      'warning'
+    )
+    handleClose()
+  }
+
+  const copyWithoutPass = async () => {
+    await props.copyToClipboard(
+      url,
+      'Private room URL without password copied to clipboard',
+      'success'
+    )
+    handleClose()
+  }
+
+  const showIf = (when: boolean) => {
+    return { display: when ? '' : 'none' }
+  }
```

Comment:
Rather than hiding UI with CSS, what do you think about relying on conditional rendering? For example, with the `DialogTitle` below:

```tsx
{isAdvanced ? )
  <DialogTitle id="alert-dialog-title">
    Copy URL with password
    <Button onClick={() => setIsAdvanced(false)}>Simple</Button>
  </DialogTitle>
) : null}
```

Reply:
yup, that's much nicer